### DISCUSSION TITLE
Saving to database as unique value

### DISCUSSION THREAD

**Discussion by LeftPinkie** [deligoez/laravel-model-hashid#27] [Q&A] [answered]
I would like to save the hash ID to the database in a unique field... Does this library check if a hash ID already exist in the database and attempt to generate another one? If so how many attempts will it make before ending with an error?

**Comment by ellisio** [comment]
No, it does not do any collision detection.

https://github.com/deligoez/laravel-model-hashid/blob/main/src/Traits/SavesHashId.php#L18-L29

**Comment by deligoez** [comment, marked as answer]
To clarify: Hash IDs are **deterministic** — the same model key always produces the same hash ID (given the same salt, length, and alphabet). So there is no collision risk when saving to the database. If your primary keys are unique (which they are), the generated hash IDs will also be unique.

In **v4.0.0**, we've added a `Blueprint::hashId()` macro that creates a `nullable`, `unique` string column for you:

```php
Schema::create("users", function (Blueprint $table) {
    $table->id();
    $table->hashId(); // nullable, unique string column
    $table->timestamps();
});
```

See the [Blueprint Macro](https://github.com/deligoez/laravel-model-hashid#blueprint-macro) section in the README.
